Modules/Devices/src/Devices.Application/Devices/Queries/ListDevices/Handler.cs

```diff
@@ -10,23 +9,17 @@ namespace Backbone.Modules.Devices.Application.Devices.Queries.ListDevices;
 public class Handler : IRequestHandler<ListDevicesQuery, ListDevicesResponse>
 {
     private readonly IdentityAddress _activeIdentity;
-    private readonly IMapper _mapper;
     private readonly IIdentitiesRepository _identitiesRepository;
 
-    public Handler(IMapper mapper, IUserContext userContext, IIdentitiesRepository devicesRepository)
+    public Handler(IUserContext userContext, IIdentitiesRepository devicesRepository)
     {
-        _mapper = mapper;
         _activeIdentity = userContext.GetAddress();
         _identitiesRepository = devicesRepository;
     }
 
     public async Task<ListDevicesResponse> Handle(ListDevicesQuery request, CancellationToken cancellationToken)
     {
-
-        var dbPaginationResult = await _identitiesRepository.FindAllDevicesOfIdentity(_activeIdentity, request.Ids, request.PaginationFilter, cancellationToken);
-
-        var items = _mapper.Map<DeviceDTO[]>(dbPaginationResult.ItemsOnPage);
-
-        return new ListDevicesResponse(items, request.PaginationFilter, dbPaginationResult.TotalNumberOfItems);
+        var dbPaginationResult = await _identitiesRepository.FindAllDevicesOfIdentity(_activeIdentity, request.Ids.Select(DeviceId.Parse), request.PaginationFilter, cancellationToken);
+        return new ListDevicesResponse(dbPaginationResult.ItemsOnPage.Select(d => new DeviceDTO(d)), request.PaginationFilter, dbPaginationResult.TotalNumberOfItems);
```

Comment:
Let the `ListDevicesResponse` do the mapping (same for all the other cases across the codebase)
```suggestion
        return new ListDevicesResponse(dbPaginationResult, request.PaginationFilter);
```